src/main/java/org/n52/wacodis/javaps/preprocessing/graph/TrainDataOperator.java

```diff
@@ -0,0 +1,119 @@
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.PropertyDescriptor;
+
+/**
+ *
+ * @author LukasButzmann
+ */
+public class TrainDataOperator extends InputDataOperator<SimpleFeatureCollection> {
+
+    private String category;
+    private String attributeName;
+
+    public TrainDataOperator(String attributeName) {
+        this.attributeName = attributeName;
+        this.category = "class";
+    }
+    
+    public TrainDataOperator(String attributeName, String category) {
+        this.attributeName = attributeName;
+        this.category = category;
+    }
+
+    @Override
+    public String getName() {
+        return "org.wacodis.writer.TrainDataOperator";
+    }
+
+    @Override
+    public SimpleFeatureCollection process(SimpleFeatureCollection input) throws WacodisProcessingException {
+
+        
+        if (hasDescriptor(attributeName, input)) {
+            
+            if (attributeName != category){
+                //NEW SCHEMA
+                SimpleFeatureType schema = input.getSchema();
+                // create new schema
+                SimpleFeatureTypeBuilder builder = new SimpleFeatureTypeBuilder();
+                builder.setName(schema.getName());
+                builder.setSuperType((SimpleFeatureType) schema.getSuper());
+                builder.addAll(schema.getAttributeDescriptors());
+                // add new attribute
+                builder.add(category, String.class);
+                // build new schema
+                SimpleFeatureType nSchema = builder.buildFeatureType();
+
+                // loop through features adding new attribute
+                List<SimpleFeature> features = new ArrayList<>();
+                SimpleFeatureIterator iterator = input.features();
+                Map<Object, Integer> attributeMap = new HashMap<>();
+
+                try {
+                  int i = 1;
+                  while (iterator.hasNext()) {
+
+                    SimpleFeature f = iterator.next();
+                    Object key = f.getAttribute(attributeName);
+
+
+                    if (!attributeMap.containsKey(key)) {
+                        attributeMap.put(key, i++);
+                    }
+
+                    //copy feature
+                    //attributeMap.put(key, attributeMap.get(key));
+
+                    SimpleFeature f2 = DataUtilities.reType(nSchema, f);
+                    f2.setAttribute(category, attributeMap.get(key));
+                    features.add(f2);
+                  }
+                } finally {
+                    iterator.close();
+                }
+
+                SimpleFeatureCollection simpleCollection = DataUtilities.collection(features);
+                return simpleCollection;
+            }else{
+                return(input);
+            }
+        }else{
+            String message = "Error! The Features in InputCollection don't have the Attribute <"+attributeName+">!";
```

Comment:
No need to include "Error!" within the exception message as the exception means an error for itself. In addition, the message can be passed directly to the exception constructor without storing it within an extra variable.